python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -140,7 +139,10 @@ def dist_inference(g, gnn_encoder, node_feats, batch_size, fanout,
                     assert len(g.ntypes) == 1
                     output_nodes = {g.ntypes[0]: output_nodes}
 
-                h = {k: x[k][input_nodes[k]].to(device) for k in input_nodes.keys()}
+                if i == 0:
+                    h = get_input_embeds(blocks[0], input_nodes)
```

Comment:
Will this cause duplicated input feature computation?
I believe so.

Reply:
i will. it trades off computation for memory. if the input layer is very simple, such as MLP, this works fine. if the input layer is BERT, we probably don't want to do it.

Reply:
Shall we implement two logics to cover both use cases?

Reply:
probably not here. I'll make some modifications in the place where `dist_inference` is called.